### ISSUE TITLE
path level parameters not rendered when operation level parameters list is empty

### ISSUE THREAD

**Issue by sylecn** [swagger-api/swagger-ui#1802]
Even after Issue #1118 is fixed, the path level parameters is not rendered when operation level parameters list is empty.

Minimum example file to show the bug:
```yaml
swagger: '2.0'
info:
  title: foo
  description: bar
  version: 1.0.0
schemes:
  - https
basePath: /v1
consumes:
  - application/json
produces:
  - application/json
paths:
  /foo:
    parameters:
      - name: fooPathLevel
        in: query
        type: string
        description: foo parameter defined in path level
        required: true
    get:
      summary: foo
      description: foo
      responses:
        '200':
          description: get foo ok
    delete:
      summary: foo
      description: foo
      parameters:
        - name: fooOperationLevel
          type: string
          in: query
          description: foo
          required: false
      responses:
        '200':
          description: delete foo ok
```

In this example, fooPathLevel is not rendered by swagger-ui. Tested in pre-build version in master, and v2.1.7-M1 tag. Build v2.1.8-M1/master on my server has failed.

I think path level parameters are not rendered because the render engine think operation parameters list is empty and didn't consider path level parameters. An easy way to get around would be to insert path level parameters to every operation level parameters list. As shown below.

```diff
diff --git a/dist/swagger-ui.js b/dist/swagger-ui.js
index c15d166..f1ba2fd 100644
--- a/dist/swagger-ui.js
+++ b/dist/swagger-ui.js
@@ -2072,19 +2072,21 @@ Resolver.prototype.resolve = function (spec, arg1, arg2, arg3) {
         // location = path[method];
         location = '/paths' + name;
         this.resolveInline(root, spec, path, resolutionTable, unresolvedRefs, location);
+      } else if (method === 'parameters') {
+        continue;
       }
       else {
         operation = path[method];
         var sharedParameters = path.parameters || [];
-        var parameters = operation.parameters || [];
+        operation.parameters = operation.parameters || []
 
         for (i in sharedParameters) {
           var parameter = sharedParameters[i];
-          parameters.unshift(parameter);
+          operation.parameters.unshift(parameter);
         }
 
-        for (i in parameters) {
-          var parameter = parameters[i];
+        for (i in operation.parameters) {
+          var parameter = operation.parameters[i];
           location = '/paths' + name + '/' + method + '/parameters';
 
           if (parameter.in === 'body' && parameter.schema) {
```

Thanks,
Yuanle

**Comment by fehguy** [comment]
thanks--but please test master, which is 2.1.4.  The *-M* builds are quite old.

**Comment by sylecn** [comment]
The version in master's dist/ dir is 2.1.4. I have tested that version, I can confirm it has the problem.

**Comment by fehguy** [comment]
This has been fixed in swagger-js v2.1.9 which will be released shortly, and be part of swagger-ui 2.1.4.
